src/types.rs

```diff
@@ -50,6 +50,15 @@ pub trait Cipher : Send + Sync {
     #[must_use]
     /// Decrypt (with associated data) a given ciphertext.
     fn decrypt(&self, nonce: u64, authtext: &[u8], ciphertext: &[u8], out: &mut[u8]) -> Result<usize, ()>;
+
+    /// Rekey according to Section 4.2 of the Noise Specification, with a default
+    /// implementation guaranteed to be secure for all ciphers.
+    fn rekey(&mut self) {
+        let mut ciphertext = [0; 32 + TAGLEN];
```

Comment:
maybe time to make a `CIPHERKEYLEN` for consistency (although really all these constants should have underscores shouldn't they - feel free to change all of them in a separate commit if you feel so inclined).

Reply:
fixed in 056e9770d8f5